### DISCUSSION TITLE
Hiding the External CMD Console for a Cleaner ComfyUI Experience

### DISCUSSION THREAD

**Discussion by nicarinai** [comfyanonymous/ComfyUI#6101] [Ideas]
Since the UI update, ComfyUI now allows operating the CMD window internally using the `Toggle Bottom Panel`.  
As a result, I think the CMD window that opens at the start of ComfyUI is no longer necessary.  
This window only adds to the clutter, getting buried among many programs in the taskbar.  

To address this, I created a `.vbs` script to hide the console window.  
You can use this `.vbs` file instead of the `.bat` file you used to run ComfyUI previously.  

However, one inconvenience is that to shut down ComfyUI, you’ll have to open the Task Manager (`Ctrl+Shift+ESC`) and manually terminate `python.exe`.  
Since modifying the original code could result in issues like code conflicts or resets with updates, creating a `.bat` file to terminate all instances of `python.exe` seems to be the best solution for now.

Below is the code to hide the console window:

```vbs
Set WshShell = CreateObject("WScript.Shell") 
WshShell.Run """C:\Users\Username\Documents\ComfyUI\ComfyUI_Launcher.bat""", 0, False
WshShell.Popup "Start ComfyUI", 5, "ComfyUI", 64
```

The `WshShell.Run` command points to the path of the `.bat` file previously used to launch ComfyUI.  
You should replace the path `C:\Users\Username\Documents\ComfyUI\ComfyUI_Launcher.bat` with one that matches your environment.  

The `WshShell.Popup` command is optional, but I included it because some feedback is needed during the loading process.  

Here is the code for shutting down ComfyUI:

```bat
@echo off
taskkill /im python.exe /f
```
*This bat file kills every instance of python.exe. If you are working with multiple Python applications, you might not want to use this.

Save each code snippet in a text editor with the extensions `.vbs` and `.bat`, respectively.  

I believe there might be a cleaner solution, but since I’m not a programmer, I’ve done my best by researching and trying to make it as clean as possible. I’m sharing this in case it’s useful for others.

**Comment by andreszs** [comment]
It would be weird to see the comfyui log inside the browser, I didn't even know you could do that. 

Now, because the user may have **different tabs open with different ComfyUI workflows**, having one CMD to rule them all is essential. The CMD taskbar icon doesn't bother me at all, there's plenty of space in the Windows 10 taskbar on a 1920x1080 resolution and I don't pin everything to the taskbar "just in case", I still prefer to launch stuff from the start menu tiles, meaning my taskbar is kept clean at all times. (I understand the tiles are gone from the Windows 11 start menu, but the fact that user's don't know how to arrange their shortcuts it's not our problem anyway)
